src/testing/snaptest.zig

```diff
@@ -176,6 +192,64 @@ pub const Snap = struct {
     }
 };
 
+fn equal_excluding_ignored(got: []const u8, snapshot: []const u8) bool {
+    var got_rest = got;
+    var snapshot_rest = snapshot;
+
+    // Don't allow ignoring suffixes and prefixes, as that makes it easy to miss trailing or leading
+    // data.
+    assert(!std.mem.startsWith(u8, snapshot, "<snap:ignore>"));
+    assert(!std.mem.endsWith(u8, snapshot, "<snap:ignore>"));
+
+    for (0..10) |_| {
+        // Cut the part before the first ignore, it should be equal between two strings...
+        const snapshot_cut = stdx.cut(snapshot_rest, "<snap:ignore>") orelse break;
+        const got_cut = stdx.cut(got_rest, snapshot_cut.prefix) orelse return false;
+        if (got_cut.prefix.len != 0) return false;
+        got_rest = got_cut.suffix;
+        snapshot_rest = snapshot_cut.suffix;
+
+        // ...then find the next part that should match, and cut up to that.
+        const next_match = if (stdx.cut(snapshot_rest, "<snap:ignore>")) |snapshot_cut_next|
+            snapshot_cut_next.prefix
+        else
+            snapshot_rest;
+        assert(next_match.len > 0);
+        snapshot_rest = stdx.cut(snapshot_rest, next_match).?.suffix;
+
+        const got_cut_next = stdx.cut(got_rest, next_match) orelse return false;
+        const ignored = got_cut_next.prefix;
+        // If <snap:ignore> matched an empty string, report it as an error.
+        if (ignored.len == 0) return false;
```

Comment:
Maybe also error if the ignored string contains a newline? Since that would probably be confusing and/or a mistake.

Reply:
Good idea! I had the same one, but forgot to act on it :rofl: